src/test/java/com/ocient/transport/NettyOcientWireV1TransportTest.java

```diff
@@ -0,0 +1,128 @@
+package com.ocient.transport;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.nio.charset.Charset;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.DefaultChannelPipeline;
+import io.netty.channel.embedded.EmbeddedChannel;
+
+@SuppressWarnings("unused")
+@RunWith(MockitoJUnitRunner.class)
+public class NettyOcientWireV1TransportTest {
+
+    private NettyOcientWireV1Transport transport;
+
+    @Mock
+    private ChannelFuture initChannelFuture;
+
+    @Mock
+    private ChannelFuture writeChannelFuture;
+
+    @Spy
+    private Channel channel;
+
+    @Spy
+    private ChannelPipeline pipeline;
+
+    @Before
+    public void setUp() {
+
+        doAnswer(invocation -> {
+            System.out.println("Get channel");
+            return this.channel;
+        })
+            .when(this.initChannelFuture)
+            .channel();
+
+        doAnswer(invocation -> {
+            System.out.println("Get pipeline");
+            return this.pipeline;
+        })
+            .when(this.channel)
+            .pipeline();
+
+        doAnswer(invocation -> {
+            System.out.println("addLastCalled");
+            return null;
```

Comment:
```suggestion
            return this.pipeline;
```

I would also recommend setting a breakpoint on the `NullPointerException' default constructor and running in debug mode.